libcontainer/standard_init_linux.go

```diff
@@ -79,6 +79,14 @@ func (l *linuxStandardInit) Init() error {
 	}
 
 	label.Init()
+
+	// Use Unshare() to enter a new cgroup_namespace before InitializeMountNamespace()
+	if l.config.Config.Namespaces.Contains(configs.NEWCGROUP) && l.config.Config.Namespaces.PathOf(configs.NEWCGROUP) == "" {
+		if err := syscall.Unshare(configs.Syscall_CLONE_NEWCGROUP); err != nil {
```

Comment:
OK, I also have considered the way of using pipe to block `nsexec.c` code but got stuck someway, so I just sent this implementation for discussion. Though I would try to work it out.